textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -37,9 +40,10 @@
     private boolean spanSchemaDefined = false;
     private Schema spanSchema;
 
-    public KeywordMatcher(IPredicate predicate, ISourceOperator sourceOperator) {
+    public KeywordMatcher(IPredicate predicate) {
         this.predicate = (KeywordPredicate)predicate;
-        this.sourceOperator = sourceOperator;
+        DataReaderPredicate dataReaderPredicate = this.predicate.convertToDataReaderPredicate();
+        this.sourceOperator = new IndexBasedSourceOperator(dataReaderPredicate);
```

Comment:
Because this.predicate is keyword Predicate, and the IndexBasedSourceOperator take a DataReaderPredicate as input. 

Proper design would have been to inherit all the other predicates for DataReaderPredicate, and avoid the conversion at all the places. Since it was design decision, I didn't want change anything without consulting. And also it should be consistent with all other operators are doing.